--- vboxnetadp/linux/VBoxNetAdp-linux.c.orig	2016-12-15 12:56:59.054184354 -0500
+++ vboxnetadp/linux/VBoxNetAdp-linux.c	2016-12-15 12:57:18.345184041 -0500
@@ -151,8 +151,6 @@
     /* Update the stats. */
     pPriv->Stats.tx_packets++;
     pPriv->Stats.tx_bytes += pSkb->len;
-    /* Update transmission time stamp. */
-    pNetDev->trans_start = jiffies;
     /* Nothing else to do, just free the sk_buff. */
     dev_kfree_skb(pSkb);
     return 0;
